components/blitz/src/omero/gateway/facility/TablesFacility.java

```diff
@@ -263,6 +268,9 @@ public TableData getTable(SecurityContext ctx, long fileId, long rowFrom,
     public long[] query(SecurityContext ctx, long fileId, String condition,
             long start, long stop, long step) throws DSOutOfServiceException,
             DSAccessException {
+        if (fileId < 0)
```

Comment:
Other methods return ``null`` when the id is ``<0``
why is it different for that method?

Reply:
That's true, returning `null` or even throw an IllegalArgumentException would be more consistent.